spec.md

```diff
@@ -8,121 +8,162 @@ title: Semantic Specification
     [Span C········································]
-         [Span E···········]   [Span F········]
-                                [Span G······]
-                                [Span H··]
-                                        [Span I·]
+         [Span E·······]        [Span F··] [Span G··] [Span H··]
 ~~~
 
 ## Traces
 
-A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system. A Trace can be thought of as a tree of Spans. (See the ASCII diagrams above)
+A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system. A Trace can be thought of as a directed acyclic graph (DAG) of Spans.
 
 ## Spans
 
-A **Span** represents a logical unit of work in the system that has a start time and a duration. Spans may be nested and ordered to model parent-child and causal relationships. Each span has an **operation name**, a presumably human-readable string which concisely names the work done by the span (e.g., an RPC method name, a function name, or the name of a subtask within a larger computation).
+A **Span** represents a logical unit of work in the system that has a start time and a duration. Spans may be nested and ordered to model causal relationships. Each span has an **operation name**, a presumably human-readable string which concisely represents the work done by the span (e.g., an RPC method name, a function name, or the name of a subtask or stage within a larger computation).
+
+<span id="references"></span>
+
+### Causal Span References
+
+A Span may reference zero or more Spans that are causally related. OpenTracing presently defines two types of references:
+
+**Parent references:** A "parent" Span may create a new "child" Span. The expectation is that the parent Span has not finished by the time the new Span starts. Though it is often the case that the parent Span cannot complete successfully until the child Span does as well, this is not a semantic requirement for parent references.
+
+A timing diagram for a parent that's blocked on a child Span:
+
+~~~
+    [-Parent Span---------]
+         [-Child Span----]
+~~~
+
+A timing diagram for a parent that's not blocked on a child Span:
+
+~~~
+    [-Parent Span-]
+              [-Child Span-]
+~~~
+
+**Sequence predecessor references:** Some Spans cannot start until a "sequence predecessor" Span finishes. For instance, chained sequence predecessor references neatly describe pipeline stages separated by queues.
+
+A sequence predecessor Span is part of the same logical trace as the descendant Span that refers to it: i.e., the descendant Span is formally caused by the work of its sequence predecessor.
+
+A timing diagram for a SequencePredecessorRef that must finish before the descendant Span can start:
+
+~~~
+    [-SequencePredecessorRef Span-]  [-New Span-]
+~~~
+
 
```

Comment:
@dawallin To me, the quintessential "fire-and-forget" request is a cache insert... no need to block the application path, and no need to do anything (aside from log failure) if it doesn't work. A pipeline is different, of course... there are typically queues (kafka, SQS, or even in-memory queues, whatever) between stages, and the application logic breaks if stages don't happen or fail (unlike fire-and-forget cache inserts, etc).

Also, the term "blocking" was removed from the Go PR, mainly for these reasons.

At issue here is our desire to keep the number of reference types as small as we can (since it's easier to add them than to remove them) while still adding value via these reference types. In my experience, the sequential pipeline case is the most important one to distinguish... it has its own heuristics from a UI/presentation standpoint as well as from a trace assembly and critical path standpoint. So it's the one I think we can afford to split off. As we gain more experience with reference types, perhaps we can split off common patterns into their own smaller types/subtypes of the Parent and SequentialPredecessor reference types.

Thoughts?